lib/telegraf/sidekiq.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'rack'
+
+module Telegraf
+  module Sidekiq
+    # Telegraf::Sidekiq::Middleware
+    #
+    # This Sidekiq middleware collects queue metrics and sends them to telegraf.
+    #
+    #
+    # Tags:
+    #
+    # * `type`:
+    #     One of "job" or "scheduled_job".
+    #
+    # * `queue`:
+    #     The queue this job landed on.
+    #
+    # * `worker`:
+    #     The name of the worker class that was executed.
+    #
+    # * `errors`:
+    #     Whether or not this job errored.
+    #
+    # * `retry`:
+    #     Whether or not this execution was a retry of a previously failed one.
+    #
+    #
+    # Values:
+    #
+    # * `app_ms`:
+    #     Total worker processing time.
+    #
+    # * `queue_ms`:
+    #     How long did this job wait in the queue before being processed?
+    #     Only present for "normal" (async) jobs (with tag `type` of "job").
+    #
+    class Middleware
+      def initialize(agent:, series: 'sidekiq', tags: {})
+        @agent = agent
+        @series = series.to_s.freeze
+        @tags = tags.freeze
+      end
+
+      def call(worker, job, queue)
+        job_start = ::Time.now.utc
+
+        tags = {
+          **@tags,
+          type: 'job',
+          errors: true,
+          retry: job.key?('retried_at'),
```

Comment:
I tried very hard to test retries, but that is not possible, as far as I could tell (and what Sidekiq's author commented in a question on StackOverflow).

Reply:
Fine for me.